### ISSUE TITLE
Project Infrastructure

### ISSUE THREAD

**Issue by PseudoNerd** [EnigmaVSSUT/oktober-project#1]
### Items to be discussed for the Hacktoberfest project before 21st September.

- [ ] **Discussion of Project ideas before the commencement of Hacktoberfest.**
- [ ] **Deciding the frameworks and tools to be used and implemented**
- [ ] **Use of third party applications** 
- [ ] **Project timeline**  
- [ ] **Division of responsibilities**

Let the discussion begin. 

[Idea list](https://github.com/EnigmaVSSUT/oktober-project/issues/1#issuecomment-687824282)

**Comment by PseudoNerd** [comment]
Please leave your ideas on this thread.

**Comment by soum-sr** [comment]
- IDEA_1:- Building some machine learning or deep learning application and deploying them to the web using Django. Then using ReactJS to link with Django to handle the front end. 
- IDEA_2:- Creating a python package for some things that we mostly do in python but require the help of multiple packages. In layman terms: Just copy and pasta the existing features from different libraries then combining them to make something powerful and effective. (PLUS we can add our own flavor to the pasta)
- IDEA_3:- IDEA_1, instead of using Django, we'll use Flutter (for mobile apps) and understand how to deploy ml applications (face detection as a starter) directly on the device.

**Comment by PseudoNerd** [comment]
> * IDEA_1:- Building some machine learning or deep learning application and deploying them to the web using Django. Then linking ReactJS to link with Django to handle the front end.

This sounds really good @soum-sr.

Touches almost every base of programming in Enigma. 
What say, @msatul1305 ?

**Comment by partha2000** [comment]
<u>Ideas</u>
1. Using the Codeforces API and web scrapping on codechef make a webapp or an android app to make a public leader-board of our college students.

**Comment by msatul1305** [comment]
> > * IDEA_1:- Building some machine learning or deep learning application and deploying them to the web using Django. Then linking ReactJS to link with Django to handle the front end.
> 
> This sounds really good @soum-sr.
> 
> Touches almost every base of programming in Enigma.
> What say, @msatul1305 ?

Yeah, Definitely you can go with this one!

**Comment by PseudoNerd** [comment]
The discussion is still open and would encourage active participation by the members.

**Comment by PseudoNerd** [comment]
> Ideas
> 
> 1. Using the Codeforces API and web scrapping on codechef make a webapp or an android app to make a public leader-board of our college students.

Do you have any specific technologies on mind for the frontend and the backend ?

**Comment by Sitispeaks** [comment]
Flutter+Flask/Django might work

**Comment by tripathyprateek** [comment]
> > Ideas
> > 
> > 1. Using the Codeforces API and web scrapping on codechef make a webapp or an android app to make a public leader-board of our college students.
> 
> Do you have any specific technologies on mind for the frontend and the backend ?

We can make a simple Chrome Extension for this also. It will be easier to implement, and to add a Leaderboard, we can use Codeforces API to show all users.

**Comment by raffleberry** [comment]
> Ideas
> 
> 1. Using the Codeforces API and web scrapping on codechef make a webapp or an android app to make a public leader-board of our college students.

I support this idea.

**Comment by tripathyprateek** [comment]
Project Ideas.
1. A fully functional Job/Internship app for our college in which Companies and students can directly register, have discussion and video call and see progress of their interviews. Tech Stack - React for Frontend and Node+Rest API for backend. Might cut some of the chaos that happens during placements. 
2. Trello Clone, designed for University Students -[Example](http://codeloveandboards.com/blog/2016/01/04/trello-tribute-with-phoenix-and-react-pt-1/)
3. Making a Package for streaming purpose - [Something like this](https://github.com/webtorrent/instant.io)

**Comment by PseudoNerd** [comment]
> Project Ideas.
> 
> 1. A fully functional Job/Internship app for our college in which Companies and students can directly register, have discussion and video call and see progress of their interviews. Tech Stack - React for Frontend and Node+Rest API for backend. Might cut some of the chaos that happens during placements.
> 2. Trello Clone, designed for University Students -[Example](http://codeloveandboards.com/blog/2016/01/04/trello-tribute-with-phoenix-and-react-pt-1/)

- While this sounds great, most recruiting companies usually have their own platforms for the process. On the other hand, we could have a placement dashboard to showcase past placement data.

- The tech stack is a miss on the Pythonistas of the organization, which is why a balance is to be created between the languages and frameworks to be used. 

However, a good thought to begin with.

**Comment by partha2000** [comment]
> Project Ideas.
> 
> 1. A fully functional Job/Internship app for our college in which Companies and students can directly register, have discussion and video call and see progress of their interviews. Tech Stack - React for Frontend and Node+Rest API for backend. Might cut some of the chaos that happens during placements.
> 2. Trello Clone, designed for University Students -[Example](http://codeloveandboards.com/blog/2016/01/04/trello-tribute-with-phoenix-and-react-pt-1/)
> 3. Making a Package for streaming purpose - [Something like this](https://github.com/webtorrent/instant.io)

This will be a little difficult, to begin with, but can be done.:thumbsup: I support the first one

**Comment by tripathyprateek** [comment]
> > Project Ideas.
> > 
> > 1. A fully functional Job/Internship app for our college in which Companies and students can directly register, have discussion and video call and see progress of their interviews. Tech Stack - React for Frontend and Node+Rest API for backend. Might cut some of the chaos that happens during placements.
> > 2. Trello Clone, designed for University Students -[Example](http://codeloveandboards.com/blog/2016/01/04/trello-tribute-with-phoenix-and-react-pt-1/)
> > 3. Making a Package for streaming purpose - [Something like this](https://github.com/webtorrent/instant.io)
> 
> This will be a little difficult, to begin with, but can be done.👍 I support the first one

yeah, and instead of Node for Backend we can use DJango as well, for the Pythonistas.

**Comment by partha2000** [comment]
**_Some more Ideas_**
1. **OLX clone**: Make a simple app for our college, where one can sell old stuff and anything of interest. 
    Tech_Stack: Android(Java/ Kotlin). Payments(optional) can be handled using the Google_Pay API.
2. **A free resume generator:** User can input details and can get a nicely made resume along with some styles to choose from.
    Front_end: ReactJs ,Backend: NodeJs/ Django
3. **Free resume evaluator**: Refer to [this](https://resumeworded.com/). Give a score to a resume based on some parameters

**Comment by Sounmay** [comment]
Ideas:
1. Attendance app to replace and digitize the traditional paper based attendance system at Uni. Use a Tinder-like interface to mark present/absent through swipes. 

2. A club registry website, displaying information about all the clubs in the university. Can show the members, achievements, programmes under the clubs etc. 

3. I have been working on a Flutter food delivery app as a freelance project, built on Firebase, the code for which can be found on my latest repository. Suggestions for improvement are welcome. Moreover, i shall be working on an e-commerce website and its' flutter application in September and October. Same goes for that.

4. One of the main things lacking at Bhisut is an internal emailing system. Would love to hear ideas about building one.

5. Long term project - We could build a universal app to combine all the facets of student life. Students could log in, check their attendance, grades, placement information. We could add a side for professors to keep track of their classes, attendance registry etc. Would be great to hear takes on this.

**Comment by tripathyprateek** [comment]
> Ideas:
> 
> 1. Attendance app to replace and digitize the traditional paper based attendance system at Uni. Use a Tinder-like interface to mark present/absent through swipes.
> 2. A club registry website, displaying information about all the clubs in the university. Can show the members, achievements, programmes under the clubs etc.
> 3. I have been working on a Flutter food delivery app as a freelance project, built on Firebase, the code for which can be found on my latest repository. Suggestions for improvement are welcome. Moreover, i shall be working on an e-commerce website and its' flutter application in September and October. Same goes for that.
> 4. One of the main things lacking at Bhisut is an internal emailing system. Would love to hear ideas about building one.
> 5. Long term project - We could build a universal app to combine all the facets of student life. Students could log in, check their attendance, grades, placement information. We could add a side for professors to keep track of their classes, attendance registry etc. Would be great to hear takes on this.

Emailing system and automation is of dire need for our College. Would love to implement that.

**Comment by PseudoNerd** [comment]
> 2\. A club registry website, displaying information about all the clubs in the university. Can show the members, achievements, programmes under the clubs etc.

This sounds like an achievable project encouraging healthy participation.


> 4\. One of the main things lacking at Bhisut is an internal emailing system. Would love to hear ideas about building one.

While this sounds necessary, this requires monetary resources for implementation on a large scale. This is an excellent thing to try out but real-time feasibility seems a little low.

**Comment by PseudoNerd** [comment]
/polls "A club registry website" "Placement dashboard to showcase past placement data" ""

**Comment by PseudoNerd** [comment]
Let's start with the Club Registry Project, shall we ?

**Comment by partha2000** [comment]
Here are some more project ideas(some are same as discussed here) : https://docs.google.com/document/d/1LUnaUXM93SqC8nMCQjZYir4WRIz2RM47c_ozF_tUR4A/edit
